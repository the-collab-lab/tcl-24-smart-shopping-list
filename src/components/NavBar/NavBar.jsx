import { NavLink } from 'react-router-dom';

import { NavBarContainer } from './NavBar.style.js';

const NavBar = () => (
  <NavBarContainer className="nav-container">
    <div>
      <NavLink className="nav-link" exact to="/list">
        <i className="fas fa-list-ul"></i>
        <span>List</span>
      </NavLink>
      <NavLink className="nav-link" exact to="/addItem">
        <i className="fas fa-plus"></i>
        <span>Add item</span>
      </NavLink>
    </div>
  </NavBarContainer>
);

export default NavBar;
