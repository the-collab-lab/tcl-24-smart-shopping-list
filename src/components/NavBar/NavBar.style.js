import styled from 'styled-components';
import { colorPalet } from '../Style/Color.palette.js';

const { primary, secundary, bgApp, active, dark, bg } = colorPalet;

export const NavBarContainer = styled.nav`
  display: flex;
  position: fixed;
  bottom: 0;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  min-width: 375px;
  width: 100%;
  height: 10vh;
  background: ${bgApp};
  div {
    display: flex;
    align-items: center;
    justify-content: space-around;
    width: 30%;
    height: 3.2rem;
    border-radius: 100px 100px 100px 100px;
    font-size: 1.2rem;
    background-color: ${dark};
    border: 0.1rem solid ${dark};
    box-shadow: 0 0px 6px rgba(0, 0, 0, 0.2);
  }
  span {
    display: none;
  }
  a.nav-link {
    text-decoration: none;
    font-weight: bold;
    text-transform: uppercase;
    padding: 1.2rem 0.8rem;
    i {
      font-size: 2rem;
      color: ${bgApp};
      &:hover {
        color: ${active};
      }
    }
  }

  @media (min-width: 480px) {
    width: 420px;
    height: 10vh;
    position: relative;
    min-width: 420px;
    margin: auto;
    box-shadow: 0 0px 6px rgba(0, 0, 0, 0.2);
    border-radius: 2rem;
    div {
      width: 50%;
    }
    i {
      display: none;
    }
    span {
      display: block;
      font-size: 1.2rem;
      font-weight: bold;
      text-transform: uppercase;
      color: ${bgApp};
      &:hover {
        color: ${active};
      }
    }
  }
`;
