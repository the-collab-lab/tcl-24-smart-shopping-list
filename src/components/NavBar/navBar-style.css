.active {
  font-weight: bold;
}

a {
  margin-left: 2%;
  text-decoration: none;
}
