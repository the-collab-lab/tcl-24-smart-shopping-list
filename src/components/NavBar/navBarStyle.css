/* .nav-container {
  width: 100%;
  height: 5%;
  position: fixed;
  bottom: 0;
  display: flex;
  align-items: center;
  justify-content: center;
}

a.nav-link.active {
  font-weight: bold;
}

a.nav-link {
  margin-left: 2%;
  text-decoration: none;
} */
