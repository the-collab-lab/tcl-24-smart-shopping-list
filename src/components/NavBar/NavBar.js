import { NavLink } from 'react-router-dom';
import './navBarStyle.css';

const NavBar = () => {
  return (
    <nav className="nav-container">
      <NavLink className="nav-link" exact to="/list">
        {' '}
        List
      </NavLink>
      <NavLink className="nav-link" exact to="/addItem">
        {' '}
        Add Item
      </NavLink>
    </nav>
  );
};

export default NavBar;
