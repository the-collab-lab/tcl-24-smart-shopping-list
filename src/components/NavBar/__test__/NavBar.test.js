import { shallow } from 'enzyme';
import NavBar from '../NavBar';

describe('NavBar', () => {
  it('Deberia renderisar', () => {
    const wrapper = shallow(<NavBar />);

    expect(wrapper).toMatchSnapshot();
  });

  it('deberia tener 2 rutas', () => {
    const wrapper = shallow(<NavBar />);

    // console.log(wrapper.find('NavLink').debug())
    expect(wrapper.find('NavLink').length).toBe(2);
  });

  it('Deberia tener una ruta para /list y /addItem', () => {
    const wrapper = shallow(<NavBar />);

    expect(wrapper.find('NavLink').at(0).prop('to')).toBe('/list');
    expect(wrapper.find('NavLink').at(1).prop('to')).toBe('/addItem');
  });

  it('Las rutas deberian tener nav-link en su className', () => {
    const wrapper = shallow(<NavBar />);

    expect(wrapper.find('NavLink').at(0).prop('className')).toBe('nav-link');
    expect(wrapper.find('NavLink').at(1).prop('className')).toBe('nav-link');
  });
});
