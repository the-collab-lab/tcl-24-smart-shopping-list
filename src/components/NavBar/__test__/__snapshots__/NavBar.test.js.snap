// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Deberia renderisar 1`] = `
<nav
  className="nav-container"
>
  <NavLink
    className="nav-link"
    exact={true}
    to="/list"
  >
    List
  </NavLink>
  <NavLink
    className="nav-link"
    exact={true}
    to="/addItem"
  >
    Add Item
  </NavLink>
</nav>
`;
